PracticeGit/intlist.c

```diff
@@ -0,0 +1,99 @@
+//----------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+//------------------------------------------------------------------------------------------------
+//   intlist.c
+//   Program on Linked List functions.
+//--------------------------------------------------------------------------------------------
+#include <malloc.h>
+#include "intlist.h"
+
+// Initialize the head pointer of the linked list to NULL.
+struct node* head = NULL;
+
+struct node* CreateList (int initialData) {
+   struct node* head = (struct node*)malloc (sizeof (struct node));
+   if (head == NULL) return ERROR_MEM_ALLOC;
+   head->data = initialData;
+   head->next = NULL;
+   return head;
+}
+
+void Append (struct node* head, int newData) {
+   struct node* current = head;
+   while (current->next != NULL) current = current->next;
+   struct node* newNode = (struct node*)malloc (sizeof (struct node));
+   if (newNode == NULL) return ERROR_MEM_ALLOC;
+   newNode->data = newData;
+   newNode->next = NULL;
+   if (head == NULL) return newNode;  // If list is empty, make newNode the head
+   current->next = newNode;
+   return head;

```

Comment:
Your function signature says you're returning nothing, but you're returning error codes (which are int) and node* (which are pointers). You can only do one of these.
This same is valid throughout your code. You should either declare it to return int or not return any value at all.